### PR TITLE
CLOUDSTACK-8749: Add checks to prevent malformed/unexpected input

### PR DESCRIPTION
**Pull request by bhaisaab**
Based on @jburwell's comment on PR #718

### CODE REVIEW

**Comment by remibergsma** [comment]
LGTM

**Comment by remibergsma** [comment]
@jburwell can you review this please? Thanks!

File: plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/BridgeVifDriver.java [hunk 1]
```diff
@@ -286,7 +287,11 @@ private void deleteExitingLinkLocalRouteTable(String linkLocalBr) {
             String[] lines = parser.getLines().split("\\n");
             for (String line : lines) {
                 String[] tokens = line.split(" ");
-                if (!tokens[2].equalsIgnoreCase(linkLocalBr)) {
+                if (tokens != null && tokens.length < 2) {
+                    continue;
+                }
+                final String device = tokens[2];
+                if (!Strings.isNullOrEmpty(device) && !device.equalsIgnoreCase(linkLocalBr)) {
```
Comment (jburwell) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/BridgeVifDriver.java]: Reading through this block, it feels like lines 289-293 should be extracted to a private method:

```
static parseDevice(final String line) {
     final String[] tokens = line.split( " ");
     if (tokens != null && tokens.length < 2) {
        return "";
     }
     
     final String device = tokens[2];
     return Strings.isNullOrEmpty(device) : " " ? device;
} 
```

The extraction of this method would not only shorten the deleteExitingLinkLocalLocalRoutetTable method, but clearly express the intent.  It also allows a unit test to be written to verify that parsing works as expected for different types of input.

**Comment by jburwell** [comment]
While it is a small nit and not expressly within the scope of this PR, could you fix the spelling mistake in the ``deleteExitingLinkLocalRouteTable`` name?

File: plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/BridgeVifDriver.java (hunk 1 above)
Comment (bhaisaab) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/BridgeVifDriver.java]: With respect to this specific code, I would like do this the next time we do a round of refactoring, possibly factor such code to a util class that can be used as a general purpose line parsing static method and add unit tests. For now, at least the code takes care of malformed inputs.

**Comment by bhaisaab** [comment]
renamed the method, s/deleteExitingLinkLocalRouteTable/deleteExistingLinkLocalRouteTable/g

**Pull request merged**